sqllogictest/src/parser.rs

```diff
@@ -603,11 +614,85 @@ fn parse_file_inner(loc: Location) -> Result<Vec<Record>, ParseError> {
 
 #[cfg(test)]
 mod tests {
-    use crate::parse_file;
+    use difference::{Changeset, Difference};
+
+    use super::*;
 
     #[test]
     fn test_include_glob() {
         let records = parse_file("../examples/include/include_1.slt").unwrap();
-        assert_eq!(14, records.len());
+        assert_eq!(16, records.len());
+    }
+
+    #[test]
+    fn test_basic() {
+        parse_roundtrip("../examples/basic/basic.slt")
+    }
+
+    /// Parses the specified file into Records, and ensures the
+    /// results of unparsing them are the same
```

Comment:
I guess current implementation is still not enough to ensure `unparse(parse())` is `identity`,
because spaces can be anywhere in the records, not only between records.

e.g.,
```
    # aa
  query    II
...
```
Currently it will be unparsed to 
```
# aa
query II
```

But do we really need identical reconstruction (It can be hard)? If not, maybe we can just test `parse(unparse)` is the same record?

Reply:
But preserving newlines makes some sense.

Reply:
> But do we really need identical reconstruction (It can be hard)? If not, maybe we can just test parse(unparse) is the same record?

I don't really need exact reconstruction -- what I really want is to be able to "update" a file in place with expected output. If part of that update also ends up "normalizing" some of the commands (like collapsing `query    III`  `query III`) that would be fine with me

Reply:
> If not, maybe we can just test parse(unparse) is the same record?

Testing parse/unparse is a great idea -- I will add that